include/ezc3dConfig.h.in

```diff
@@ -1,12 +1,21 @@
 #ifndef EZC3D_CONFIG_H
 #define EZC3D_CONFIG_H
 
+#cmakedefine EZC3D_SHARED
+
 // dllexport/import declaration
-#ifdef _WIN32
-    #ifdef EZC3D_API_EXPORTS
-        #define EZC3D_API __declspec(dllexport)
+#ifdef EZC3D_SHARED
+    #ifdef _WIN32
+        #ifdef EZC3D_API_EXPORTS
+            #define EZC3D_API __declspec(dllexport)
+        #else
+            #define EZC3D_API __declspec(dllimport)
+        #endif
     #else
-        #define EZC3D_API __declspec(dllimport)
+        #ifdef SWIG
+            #define __attribute__(x)
+        #endif
```

Comment:
I believe it should go into .i files before including ezc3d files. C++ lib shouldn't know about swig.